flytepropeller/pkg/controller/config/config.go

```diff
@@ -258,6 +261,29 @@ type EventConfig struct {
 	FallbackToOutputReference bool            `json:"fallback-to-output-reference" pflag:",Whether output data should be sent by reference when it is too large to be sent inline in execution events."`
 }
 
+// ParallelismBehavior defines how ArrayNode should handle subNode parallelism by default
+type ParallelismBehavior = string
+
+const (
+	// ParallelismBehaviorHybrid means that ArrayNode will adhere to the parallelism defined in the
+	// ArrayNode exactly. This means `nil` will use the workflow parallelism, and 0 will have
+	// unlimited parallelism.
+	ParallelismBehaviorHybrid ParallelismBehavior = "hybrid"
+
+	// ParallelismBehaviorUnlimited means that ArrayNode subNodes will be evaluated with unlimited
+	// parallelism for both nil and 0.
```

Comment:
nit: would it provide more clarity to update this comment with something like:

"If a non-nil and non-zero parallelism is defined, then ArrayNode will adhere to the parallelism defined."

to both unlimited and workflow descriptions

Reply:
good call, just added.